### PR TITLE
Make level 8 in autopark project.

### PR DESCRIPTION
**Pull request by aces47**
Add MechanicOrders class thats working with orders.
Add orders.csv file that store information about orders.

### CODE REVIEW

**Review by dzmitri-litvinov — APPROVED** [review]

File: AutoParkConsole/Classes/VehicleType.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AutoParkConsole.Classes
+{
+	public class VehicleType
+	{
+		int _Id;
+		string _TypeName;
+		double _TaxCoefficient;
```
Comment (UNDogMan) [inline AutoParkConsole/Classes/VehicleType.cs:13]: Always set accesses modifiers, even if you now that its necessary.
Also private fields names like _fieldName

File: AutoParkConsole/Classes/Vehicle.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			double otherTaxCoefficient = other._VehicleType.GetTaxCoefficient();
+			if (thisTaxCoefficient < otherTaxCoefficient)
+			{
+				return -1;
+			}
+			else
+			{
+				return thisTaxCoefficient == otherTaxCoefficient ? 0 : 1;
+			}
+		}
```
Comment (UNDogMan) [inline AutoParkConsole/Classes/Vehicle.cs:125]: U can use doubles CompareTo method for this function.

File: AutoParkConsole/Classes/Vehicle.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	public enum Color
+	{
+		Black = 1,
+		White,
+		Red,
+		Gray,
+		Yellow,
+		Blue,
+		Green
+	}
```
Comment (UNDogMan) [inline AutoParkConsole/Classes/Vehicle.cs:15]: Put enum in separate file.

File: AutoParkConsole/Classes/MyStack.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+		void Resize()
+		{
+			T[] newStack = null;
+
+			newStack = new T[_Capacity * 2];
+			for (int counter = 0; counter < _Stack.Length; counter++)
+			{
+				newStack[counter] = _Stack[counter];
+			}
```
Comment (UNDogMan) [inline AutoParkConsole/Classes/MyStack.cs:44]: U can use CopyTo method instead of loop.

File: AutoParkConsole/Classes/MechanicOrders.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+					{
+						if(!_Orders.TryAdd(row, 1))
+						{
+							_Orders[row]++;
+						}
+					}
+				}
+			}
+		}
+
```
Comment (UNDogMan) [inline AutoParkConsole/Classes/MechanicOrders.cs:33]: Bad naming

**Review by UNDogMan — COMMENTED** [review]
Clean your code with [this](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions) or [this](https://google.github.io/styleguide/csharp-style.html) code confentions.

**Review by UNDogMan — CHANGES_REQUESTED** [review]
Check my comment.

File: AutoParkConsole/Program.cs [hunk 6]
```diff
@@ -22,24 +22,8 @@ static void Main(string[] args)
...
-			}
-
-			Console.WriteLine("Vehicles leaving the garage:");
-			int stackCount = stack.Count();
-			for (int i = 0; i < stackCount; i++)
-			{
-				Vehicle vehicle = stack.Pop();
-				Console.WriteLine($"Automobile {vehicle.ModelName} leaved");
-			}
+			MechanicOrders orders = new MechanicOrders(pathToCSV + "orders.csv");
```
Comment (PushMeNow) [inline AutoParkConsole/Program.cs:25]: I don't have this path in my file system... Use copying files in output dir on compilation and virtual path.

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: AutoParkConsole/Classes/MechanicOrders.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AutoParkConsole.Classes
+{
+	internal class MechanicOrders
+	{
+		private Dictionary<string, int> _orders;
```
Comment (PushMeNow) [inline AutoParkConsole/Classes/MechanicOrders.cs:12]: readonly

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: AutoParkConsole/Classes/MechanicOrders.cs (hunk 7 above)
Comment (aces47) [reply AutoParkConsole/Classes/MechanicOrders.cs]: Fixed

File: AutoParkConsole/Program.cs (hunk 6 above)
Comment (aces47) [reply AutoParkConsole/Program.cs]: Fixed

**Review by PushMeNow — APPROVED** [review]

File: AutoParkConsole/Program.cs [hunk 8]
```diff
@@ -1,45 +1,30 @@
...
-				2020,
-				100,
-				Color.Black,
-				150));
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string pathToCSV = Directory.GetCurrentDirectory();
+            Collections collections =
```
Comment (kuzmiich) [inline AutoParkConsole/Program.cs:12]: i 
```suggestion
            var collections =
```

**Review by kuzmiich — APPROVED** [review]
You may add var in Main method
```
Collections collections =  new Collections(pathToCSV + "types.csv", pathToCSV + "vehicles.csv", pathToCSV + "rents.csv");
MyStack<Vehicle> stack = new MyStack<Vehicle>();
```

**Review by kuzmiich — APPROVED** [review]

**Pull request merged**